Gemfile

```diff
@@ -84,12 +84,20 @@ if RUBY_PLATFORM != 'java'
   end
 end
 
-# For type checking
-# Sorbet releases almost daily, with new checks introduced that can make a
-# previously-passing codebase start failing. Thus, we need to lock to a specific
-# version and bump it from time to time.
-# Also, there's no support for windows
-if RUBY_VERSION >= '2.4.0' && (RUBY_PLATFORM =~ /^x86_64-(darwin|linux)/)
-  gem 'sorbet', '= 0.5.9672'
-  gem 'spoom', '~> 1.1'
+group :check do
+  # For type checking
+  # Sorbet releases almost daily, with new checks introduced that can make a
+  # previously-passing codebase start failing. Thus, we need to lock to a specific
+  # version and bump it from time to time.
+  # Also, there's no support for windows
+  if RUBY_VERSION >= '2.4.0' && (RUBY_PLATFORM =~ /^x86_64-(darwin|linux)/)
+    gem 'sorbet', '= 0.5.9672'
+    gem 'spoom', '~> 1.1'
+  end
+
+  # type checking with steep
+  if RUBY_VERSION >= '2.6.0' && RUBY_PLATFORM != 'java'
```

Comment:
Just out of curiosity -- does steep not run on JRuby?

Reply:
I'm honestly not sure. I don't think it matters either as it's static analysis, like Rubocop.